dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -176,6 +176,7 @@ public async Task<SKContext> RunAsync(ContextVariables variables, CancellationTo
             this.Log,
             cancellationToken);
 
+        this.Log.LogDebug("Executing kernel, Executing {0} functions in pipeline", pipeline.Length);
```

Comment:
I would re-structure log message a little bit. Also, instead of numeric labels of log args (i.e. `{0}`, `{1}`), please use named parameters. In case of using Application Insights (just as example), named arguments will be stored in separate `customDimensions` column with appropriate naming. 

I would also use log level `Information` so these logs will be available for customers by default.

There is already open PR regarding telemetry, please check it for reference: https://github.com/microsoft/semantic-kernel/pull/1905

```suggestion
        this.Log.LogInformation("Kernel execution started. Executing {PipelineLength} functions in pipeline", pipeline.Length);
```